Compiler/src/inferencestate.jl

```diff
@@ -320,6 +326,7 @@ mutable struct InferenceState
 
         nslots = length(src.slotflags)
         slottypes = Vector{Any}(undef, nslots)
+        bb_bailed = Bool[false for i = 1:length(cfg.blocks)]
```

Comment:
Is there any plan to refine this calculation in the future?  
Currently, `currbailed` is being updated in `typeinf_local` using pattern matching on `hd === :latestworld`, so it seems like this variable itself might not be necessary.

Reply:
It's path dependent and gets merged on joins, which is why it needs path state.